src/averageAges.js

```diff
@@ -14,14 +14,24 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+const MALE = 'm';
+const FEMALE = 'f';
+const ONE_CENTURY = 100;
+
+function getAverageAges(persons) {
+  return persons.reduce((sumAge, person) =>
+    sumAge + (person.died - person.born), 0) / persons.length;
 }
 
+function calculateMenAverageAge(people, century) {
+  const men = people.filter(person => person.sex === MALE && (century
+    ? Math.ceil(person.died / ONE_CENTURY) === century
+    : true));
```

Comment:
not the best formatting actually, i'd rather immediately check if there's century or no, and then write filter function